install/src/Cabal.hs

```diff
@@ -0,0 +1,109 @@
+module Cabal where
+
+import           Development.Shake
+import           Development.Shake.Command
+import           Development.Shake.FilePath
+import           Control.Monad
+import           Data.Maybe                               ( isNothing )
+import           Control.Monad.Extra                      ( whenMaybe )
+import           System.Directory                         ( findExecutable
+                                                          , copyFile
+                                                          )
+
+import           Version
+import           Print
+import           Env
+import           Stack
+
+
+execCabal :: CmdResult r => [String] -> Action r
+execCabal = command [] "cabal"
+
+execCabal_ :: [String] -> Action ()
+execCabal_ = command_ [] "cabal"
+
+cabalBuildData :: Action ()
+cabalBuildData = do
+  execCabal_ ["new-build", "hoogle"]
+  execCabal_ ["new-exec", "hoogle", "generate"]
+
+cabalBuildHie :: VersionNumber -> Action ()
+cabalBuildHie versionNumber = do
+  ghcPath <- getGhcPathOf versionNumber >>= \case
+    Nothing -> do
+      printInStars $ ghcVersionNotFoundFailMsg versionNumber
+      error (ghcVersionNotFoundFailMsg versionNumber)
+    Just p -> return p
+  execCabal_
+    ["new-build", "-w", ghcPath, "--write-ghc-environment-files=never"]
+
+cabalInstallHie :: VersionNumber -> Action ()
+cabalInstallHie versionNumber = do
+  localBin <- getLocalBin
+  execCabal_
+    [ "new-install"
+    , "--write-ghc-environment-files=never"
+    , "--symlink-bindir=" ++ localBin
+    , "exe:hie"
+    , "--overwrite-policy=always"
+    ]
+  liftIO $ do
+    copyFile (localBin </> "hie" <.> exe)
```

Comment:
Did you try, instead of resorting to `copyFile` using `copyFile' ... >> need ["filename"]` ?

Reply:
I did not. This seemed to complex and I do not want `shake` to get in the way. A simple `copyFile` is indeed all that is needed here.